### PR TITLE
Sanitizing switch() statements in kernel/*

### PR DESCRIPTION
**Pull request by ceolin**
This is part of MISRA-C fixes (tracked in #9893)

### CODE REVIEW

File: kernel/poll.c [hunk 1]
```diff
@@ -68,6 +68,7 @@ static inline int is_condition_met(struct k_poll_event *event, u32_t *state)
 		return 0;
 	default:
 		__ASSERT(0, "invalid event type (0x%x)\n", event->type);
+		break;
```
Comment (andyross) [inline kernel/poll.c]: MISRA C really demands the presence of a break after a default: case where it's a noop?  Sigh.  I mean, harmless and all, but still...

File: kernel/userspace.c [hunk 2]
```diff
@@ -342,6 +350,7 @@ static void unref_check(struct _k_object *ko)
 		k_stack_cleanup((struct k_stack *)ko->name);
 		break;
 	default:
+		/* Nothing to do */
```
Comment (andyross) [inline kernel/userspace.c]: Uh, yeah, that's what an empty case means. :)  But sure.  If you have to put the break there you might as well document what it's (not) doing.

File: scripts/gen_kobject_list.py [hunk 3]
```diff
@@ -202,14 +202,14 @@ def write_kobj_otype_output(fp):
         if dep:
             fp.write("#ifdef %s\n" % dep)
 
-        fp.write('case %s: return "%s";\n' % (kobject_to_enum(kobj), kobj))
+        fp.write('case %s: ret = "%s"; break;\n' % (kobject_to_enum(kobj), kobj))
```
Comment (andyross) [inline scripts/gen_kobject_list.py]: Too lazy to check: but the end of this generated switch statement is 100% for sure a "return ret;", right?

**Review by andyross — APPROVED** [review]
I can already tell that MISRA and I are going to get along just swimmingly...

**Review by nashif — APPROVED** [review]

**Review by ioannisg — APPROVED** [review]
Looks ok (and safe) to me.

File: kernel/poll.c (hunk 1 above)
Comment (pizi-nordic) [reply kernel/poll.c]: This break is noop only if you always put default case at the end. I think that it is safer to just a the break.

**Review by pizi-nordic — APPROVED** [review]

**Pull request merged**